bin/p2-label/main.go

```diff
@@ -0,0 +1,163 @@
+
+Example:
+    p2-label --selector $selector --add foo=bar --add bar=baz
+`).Short('a').StringMap()
+	applyDestructiveLabels = cmdApply.Flag("delete", `The list of label keys to remove from the nodes in the selector. Deletes are idempotent. Include multiple --delete switches to include multiple labels. It's safe to mix --add with --delete though the results of this command are not transactional.
+
+Example:
+  p2-label --selector $selector --delete foo --delete bar
+`).Short('d').Strings()
+
+	cmdShow       = kingpin.Command(CmdShow, "Show labels that apply to a particular entity (type, ID)")
+	showLabelType = cmdShow.Flag("labelType", "The type of label to adjust. Sometimes called the \"label tree\".Supported types can be found here:\n\thttps://godoc.org/github.com/square/p2/pkg/labels#pkg-constants").Short('t').Required().String()
+	showID        = cmdShow.Flag("id", "The ID of the entity to show labels for.").Short('i').Required().String()
+
+	// autoConfirm captures the confirmation desire abstractly across commands
+	autoConfirm = false
+)
+
+const (
+	CmdApply = "apply"
+	CmdShow  = "show"
+)
+
+func main() {
+	cmd, opts := flags.ParseWithConsulOptions()
+	client := kp.NewConsulClient(opts)
+	applicator := labels.NewConsulApplicator(client, 3)
+	exitCode := 0
+
+	switch cmd {
+	case CmdShow:
+		labelType, err := labels.AsType(*showLabelType)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error while parsing label type. Check the commandline.\n%v\n", err)
+			exitCode = 1
+			break
+		}
+
+		labelsForEntity, err := applicator.GetLabels(labelType, *showID)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Got error while querying labels. %v\n", err)
+			exitCode = 1
+			break
+		}
+		fmt.Printf("%s/%s: %s\n", labelType, *showID, labelsForEntity.Labels.String())
+		return
+	case CmdApply:
+		autoConfirm = *applyAutoConfirm
+
+		labelType, err := labels.AsType(*applyLabelType)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Unrecognized type %s. Check the commandline and documentation.\nhttps://godoc.org/github.com/square/p2/pkg/labels#pkg-constants\n", *applyLabelType)
+			break
+		}
+
+		subject, err := klabels.Parse(*applySubjectSelector)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error while parsing subject label. Check the syntax.\n%v\n", err)
+			break
+		}
+
+		additiveLabels := *applyAddititiveLabels
+		destructiveKeys := *applyDestructiveLabels
+
+		cachedMatch := false
+		matches, err := applicator.GetMatches(subject, labelType, cachedMatch)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error while finding label matches. Check the syntax.\n%v\n", err)
+			break
+		}
+
+		if len(additiveLabels) > 0 {
+			fmt.Printf("labels to be added: ")
+			for k, v := range additiveLabels {
+				fmt.Printf("%s=%s ", k, v)
```

Comment:
Personally, I'd convert to a `klables.Set` just for its `String()` method, so you don't have to write this yourself.

Also, you have trailing whitespace before a newline.